services/subnet_manager/src/test/java/com/futurewei/alcor/subnet/SubnetControllerTest.java

```diff
@@ -0,0 +1,215 @@
+package com.futurewei.alcor.subnet;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.junit.Assert.*;
+
+import com.futurewei.alcor.common.exception.ResourceNotFoundException;
+import com.futurewei.alcor.subnet.config.UnitTestConfig;
+import com.futurewei.alcor.subnet.entity.*;
+import com.futurewei.alcor.subnet.service.SubnetDatabaseService;
+import com.futurewei.alcor.subnet.service.SubnetService;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+
+@RunWith(SpringRunner.class)
```

Comment:
Like the UTs. This is a very good start. Keep going :-)

Reply:
Thanks so much for your help !